Lib/test/test_exceptions.py

```diff
@@ -319,15 +319,16 @@ def baz():
         check('(yield i) = 2', 1, 2)
         check('def f(*):\n  pass', 1, 7)
 
-    @unittest.skipIf(ctypes.sizeof(ctypes.c_ssize_t) > ctypes.sizeof(ctypes.c_int),
-                     "Py_ssize_t is bigger than INT_MAX, so this is unreachable")
+    @unittest.skipIf(ctypes.sizeof(ctypes.c_int) >= ctypes.sizeof(ctypes.c_ssize_t),
+                     "INT_MAX is bigger than Py_ssize_t, so this is unreachable")
     @support.requires_resource('cpu')
     @support.bigmemtest(support._2G, memuse=1.5)
     def testMemoryErrorBigSource(self):
-        # the padding length needs to be more than INT_MAX, but we can't
+        # the line length needs to be more than INT_MAX, but we can't
         # multiple a sequence by a number that doesn't fit Py_ssize_t,
         # otherwise we will get an OverflowError (see PySequence_Repeat)
-        padding = '       ' * ((ctypes.sizeof(ctypes.c_int) + 1) / 8)
+        INT_MAX = 2 ** (ctypes.sizeof(ctypes.c_int) * 8)
+        padding = '        ' * (INT_MAX // 8)
         with self.assertRaisesRegex(OverflowError, "column offset overflow"):
```

Comment:
Merge is needed for this as well. This doesn't check the regex anymore.

Reply:
Don't we want to keep the regex to make sure we are actually finding the expected error? I don't think it matters that much anyway, though.

Reply:
Windows 32-bit produces a different error.

Reply:
Yeah, the "cannot fit 'int' into an index-sized integer" one, but that's from the `' ' * 2**31` part, not what we intended, which is why this PR is now being careful with the `PySequence_Repeat` operation to not use a number bigger than `Py_ssize_t`. Am I missing something?

Reply:
Oh right. My error.

Reply:
No worries :smile: